### PR TITLE
Postgres tests

### PR DESCRIPTION
**Pull request by harsh-98**
# Description
- Change test that are using sqlite to use postgres. Related to https://github.com/waku-org/go-waku/issues/607

# Changes

- [x] Limit can't be negative for queries in postgres.
- [x] Change store test to use postgres.
- [x] Fixature for testing postgres/sqlite store test. 
- [x] Add postgres module tests.
- [x] Based on driverType, NewQueries from sqlite/postgres should be used in waku_node.  

# Tests

### CODE REVIEW

**Comment by jakubgs** [comment]
Fixed.

File: waku/persistence/store.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func (d *DBStore) cleanOlderRecords(ctx context.Context) error {
 	// Limit number of records to a max N
 	if d.maxMessages > 0 {
 		start := time.Now()
-		sqlStmt := `DELETE FROM message WHERE id IN (SELECT id FROM message ORDER BY receiverTimestamp DESC LIMIT -1 OFFSET $1)`
+		sqlStmt := `DELETE FROM message WHERE id IN (SELECT id FROM message ORDER BY receiverTimestamp DESC OFFSET $1)`
```
Comment (richard-ramos) [inline waku/persistence/store.go:227]: This does not work in SQLite. The LIMIT keyword is required when specifying an OFFSET.   If postgresql does not allow negative limits, then we need to extract this query, similar to how it was done with CreateTable in here https://github.com/waku-org/go-waku/blob/16ec22596ef3e4b1a65af1ea1f2a0429847c9f7a/waku/persistence/postgres/postgres.go#L59-L66 which has db specific implementations

**Review by richard-ramos — CHANGES_REQUESTED** [review]

File: waku/persistence/store.go (hunk 1 above)
Comment (richard-ramos) [reply waku/persistence/store.go]: This sounds like a bug!

File: ci/Jenkinsfile.tests [hunk 2]
```diff
@@ -30,6 +30,8 @@ pipeline {
 
   environment {
     TARGET  = 'tests'
+    DB_CONT  = "status-go-test-db-${env.EXECUTOR_NUMBER.toInteger() + 1}"
```
Comment (jakubgs) [inline ci/Jenkinsfile.tests:33]: Why is this calling the container `status-go-test-db-...`? Pretty sure this isn't `status-go` repo.

File: ci/Jenkinsfile.tests [hunk 3]
```diff
@@ -58,6 +60,28 @@ pipeline {
       } } }
     }
 
+    stage('postgres tests') {
```
Comment (jakubgs) [inline ci/Jenkinsfile.tests:63]: The whole Jenkinsfile is for tests, and the job is called `tests`. I don't think yet another `tests` in the stage names. Just call the stage `PostgreSQL`.

File: waku/persistence/postgres/mock.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+)
+
+// var dbUrlTemplate = "postgres://postgres@localhost:%s/%s?sslmode=disable"
+var dbUrlTemplate = "postgres://harshjain@localhost:%s/%s?sslmode=disable"
```
Comment (jakubgs) [inline waku/persistence/postgres/mock.go:11]: Pretty sure you don't want to leave this like this. Also, shouldn't the host and the user be also templateable?

**Review by jakubgs — APPROVED** [review]
Mostly looks fine.

**Review by richard-ramos — APPROVED** [review]
Very cool. 
if you decide to Squash & Merge the PR, do remember to change the name to match the format (something like `feat: postgresql tests`), or just use rebase & merge since the individual commit messages are fine :)

**Pull request merged**